superset-frontend/src/components/Datasource/DatasourceModal.tsx

```diff
@@ -96,39 +96,81 @@ const DatasourceModal: FunctionComponent<DatasourceModalProps> = ({
       currentDatasource.schema;
 
     setIsSaving(true);
-    SupersetClient.post({
-      endpoint: '/datasource/save/',
-      postPayload: {
-        data: {
-          ...currentDatasource,
-          cache_timeout:
-            currentDatasource.cache_timeout === ''
-              ? null
-              : currentDatasource.cache_timeout,
-          schema,
-          metrics: currentDatasource?.metrics?.map(
-            (metric: Record<string, unknown>) => ({
-              ...metric,
+    SupersetClient.put({
```

Comment:
Is it possible to save a new dataset in this modal or only update?

Reply:
only update, the `datasource/save` endpoint only supported updating an existing dataset, not creating a new one

Reply:
Ok, perfect.